Error when using Decimal128

Hi guys,

Curently, I work with Decimal128 supported bu mongodb since 3.4 version.
I use the last version of Jongo and mongo-drive 3.5 version.

Unfortunatly I have this error:

```
	at org.jongo.marshall.jackson.JacksonEngine.unmarshall(JacksonEngine.java:50)
	at org.jongo.ResultHandlerFactory$UnmarshallingResultHandler.map(ResultHandlerFactory.java:43)
	at org.jongo.MongoCursor.next(MongoCursor.java:46)
	at com.google.common.collect.Iterators.addAll(Iterators.java:357)
	at com.google.common.collect.Lists.newArrayList(Lists.java:147)
	at 
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Unknown element type 19
 at [Source: de.undercouch.bson4jackson.io.LittleEndianInputStream@3468ee6e; pos: 137] 	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:388)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:348)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1599)
```

With this content: 
`{ "_id" : { "$oid" : "5a34277664974b861aaf0423"} , "_class" : "com.allparts.ecommerce.catalog.data.entities.PartInfosEntity" , "price" : { "localizedPrices" : { "FR" : { "value" : { "$numberDecimal" : "30"} , "currency" : "EUR"}}} , "partType" : "IAM" , "vehicleIds" : [ ] , "compatibleIams" : [ ] , "compatibleOes" : [ ] , "partId" : 31 , "creationDateTime" : { "$date" : "2017-12-15T19:50:14.915Z"} , "createdBy" : "Unknown"}`

Thanks for your help.

Nizar.

--- a/pom.xml
+++ b/pom.xml
@@ -80,7 +80,7 @@
         <mongo.max.version>3.5.0</mongo.max.version>
         <mongo.version>3.5.0</mongo.version>
         <jackson.version>2.9.4</jackson.version>
-        <bson4jackson.version>2.7.0</bson4jackson.version>
+        <bson4jackson.version>2.9.0</bson4jackson.version>
     </properties>
 
     <dependencies>

--- a/src/main/java/org/jongo/marshall/jackson/bson4jackson/MongoBsonGenerator.java
+++ b/src/main/java/org/jongo/marshall/jackson/bson4jackson/MongoBsonGenerator.java
@@ -32,6 +32,11 @@ class MongoBsonGenerator extends BsonGenerator {
         super(jsonFeatures, bsonFeatures, out);
     }
 
+    @Override
+    public void flush() throws IOException {
+        _out.flush();
+    }
+
     public void writeNativeObjectId(org.bson.types.ObjectId objectId) throws IOException {
         _writeArrayFieldNameIfNeeded();
         _verifyValueWrite("write datetime");

